BackendTaskPilot/Program.cs

```diff
@@ -0,0 +1,38 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+ExchangeRateService service = new ExchangeRateService();
+
+// REST API to get exchange rates based on input currencies
+app.MapPost("/", service.GetExchangeRates);
+
+app.Run();
+
+public struct CnbRatesResponse
+{
+    [JsonPropertyName("rates")] public List<Rate> Rates { get; set; }
```

Comment:
- Place `JsonPropertyName` annotation on separate line to improve readability 
- Make property `init` only to prevent mutation after it's created
- Use the IReadOnlyCollection<T> to make property readonly
```suggestion
    [JsonPropertyName("rates")] 
    public IReadOnlyCollection<CnbRate> Rates { get; init; }
```